docs/reference/migration/migrate_6_0/scripting.asciidoc

```diff
@@ -12,3 +12,18 @@ Use painless instead.
 milliseconds since epoch as a `long`. The same is true for
 `doc.some_date_field[some_number]`. Use `doc.some_date_field.value.millis` to
 fetch the milliseconds since epoch if you need it.
+
+==== Scripts and Templates are separate things now
+
+Internally we've split scripts and templates apart. This shouldn't matter over
+the REST interface except for two things:
+* During the rolling upgrade from 5.x to 6.0 the REST APIs for templates aren't
+going to work properly most of the time. Don't use them.
```

Comment:
1. Yes, I mean "mustache/query templates". But also other places that templates are used (x-pack, mostly).
2. I only mean *stored* templates. Not file templates.
3. It is kind of complicated what will and what won't work. The safest thing is to not touch stored templates during the upgrade. You can use them, but getting, setting, or deleting them you shouldn't do.
4. If you have to modify the templates, sending the request to the master node *should* always work. Sending the request to other nodes will only work if they are on the same version as the master.
5. I can try to make these work during the upgrade and add the testing required to prove it but that seems like a lot of work. *Technically* it could be a followup. Though I'd thought it'd probably be ok to not touch the stored templates during the upgrade.